airflow/ui/src/pages/Pools/PoolBar.tsx

```diff
@@ -0,0 +1,109 @@
+import { stateColor } from "src/utils/stateColor";
+
+type PoolBarProps = {
+  readonly pools: Array<PoolResponse>;
+};
+
+const PoolBar = ({ pools }: PoolBarProps) => (
+  <Flex direction="column" gap={4}>
+    {pools.map((pool) => {
+      // Calculate proportions
+      const totalSlots = pool.slots;
+      const openFlex = pool.open_slots / totalSlots || 0;
+      const scheduledFlex = pool.scheduled_slots / totalSlots || 0;
+      const runningFlex = pool.running_slots / totalSlots || 0;
+      const queuedFlex = pool.queued_slots / totalSlots || 0;
+      const occupiedFlex = pool.occupied_slots / totalSlots || 0;
+      const deferredFlex = (pool.include_deferred ? pool.deferred_slots / totalSlots : 0) || 0;
+
+      return (
+        <Box
+          borderColor="border.emphasized"
+          borderRadius={8}
+          borderWidth={1}
+          key={pool.name}
+          overflow="hidden"
+        >
+          <Flex alignItems="center" bg="bg.muted" justifyContent="space-between" p={4}>
+            <VStack align="start">
+              <Text fontSize="lg" fontWeight="bold">
+                {pool.name}
+              </Text>
+              {Boolean(pool.description) ? (
+                <Text color="gray.fg" fontSize="sm">
+                  {pool.description}
+                </Text>
+              ) : undefined}
+            </VStack>
+            <Tooltip
+              content={pool.include_deferred ? "Deferred Slots Included" : "Deferred Slots Not Included"}
+            >
+              {pool.include_deferred ? <MdHourglassFull size={25} /> : <MdHourglassDisabled size={25} />}
+            </Tooltip>
+          </Flex>
+
+          <Box margin={4}>
+            <Flex bg="gray.100" borderRadius="md" h="20px" overflow="hidden" w="100%">
+              {/* Open Slots */}
+              <Tooltip content={`Open Slots: ${pool.open_slots}`}>
+                <Box bg={stateColor.success} flex={openFlex} h="100%" />
+              </Tooltip>
+
+              {/* Scheduled Slots */}
+              <Tooltip content={`Scheduled Slots: ${pool.scheduled_slots}`}>
+                <Box bg={stateColor.scheduled} flex={scheduledFlex} h="100%" />
+              </Tooltip>
+
+              {/* Running Slots */}
+              <Tooltip content={`Running Slots: ${pool.running_slots}`}>
+                <Box bg={stateColor.running} flex={runningFlex} h="100%" />
+              </Tooltip>
+
+              {/* Queued Slots */}
+              <Tooltip content={`Queued Slots: ${pool.queued_slots}`}>
+                <Box bg={stateColor.queued} flex={queuedFlex} h="100%" />
+              </Tooltip>
+
+              {/* Occupied Slots */}
+              <Tooltip content={`Occupied Slots: ${pool.occupied_slots}`}>
+                <Box bg={stateColor.up_for_retry} flex={occupiedFlex} h="100%" />
+              </Tooltip>
+
+              {/* Deferred Slots */}
+              {pool.include_deferred && pool.deferred_slots > 0 ? (
```

Comment:
```suggestion
              {pool.include_deferred ? (
```

If it's 0 then nothing really renders anyway.